CONTRIBUTING.md

```diff
@@ -0,0 +1,108 @@
+# CorisxTH Contributing Guidelines #
+
+## Tell other devs what you are working on! ##
+
+When you start working on a fix/enhancement for an open issue please always
+post a comment in this issue's discussion to tell other devs that you have
+started working on it so that they won't waste any of their free time by
+working on their own fix/enhancement for it.
+
+If there isn't an open issue discussion for a bug fix/enhancement you want
+to work on please check that there isn't a closed discussion for it because
+a previous developer of it didn't finish their work which you could finish.
+
+And if there is no existing issue discussion for what you want to work on
+please open a new issue for it and tell other devs here that your going to/have started working on it.
+
+**If you have an idea for an enhancement that does not have a discussion, please make a *Feature Request* first to discuss its desirability.  This helps make sure your development time is a worthy endeavour.**
+
+## Enhancement/Suggestions ##
+
+If you want to make a suggestion, please use the *Feature Request* option when making a new issue. It will provide all the information needed to evaluate the suggestion or enhancement.\
+![image](https://user-images.githubusercontent.com/20030128/123638352-fc609400-d816-11eb-88ad-11030e041fc4.png)
+
+
+## Pull Requests ##
+
+When providing a pull request ensure the description field describes the major
+changes from the request. If your pull request relates to any existing
+issues, include those issues in the description. The pull request provides a default layout to help you fill out this information.
+
+#### Pull Request Statuses ####
+CorsixTH encourages using the following labels inside your Pull Request's title to denote its current progress. Unlabelled pull request will be considered *Ready* and may be merged without warning.
+| Label | Description |
+| ----- | ----------- |
+| **[WIP]** | Work in Progress |
+| **[RFC]** | Request for Comments (aka Ready for Review) |
```

Comment:
Note that you do NOT need to RFC a PR before moving it to ready. The purpose of RFC is to get feedback on a general direction, or ask for assistance before finishing the PR. Ideally a comment would be added at the same time with the question / focus you'd like us to review.